app/main/views/sign_out.py

```diff
@@ -26,11 +26,19 @@ def _sign_out_at_login_dot_gov():
 @main.route("/sign-out", methods=(["GET", "POST"]))
 def sign_out():
     # An AnonymousUser does not have an id
-    current_app.logger.info("HIT THE REGULAR SIGN OUT")
+
     if current_user.is_authenticated:
         # TODO This doesn't work yet, due to problems above.
+        current_user.deactivate()
```

Comment:
Interesting that we have to call this explicitly!  `current_user.sign_out()` doesn't take care of this for us?

Reply:
yeah, if it did there would be no issue.  And yet there is.

If you look at the code for current_user.sign_out() the only interesting thing it does is:   

 self.update(current_session_id=None)
 
 However, trying to check current_user.current_session_id, which was an earlier iteration of what I did, was totally unreliable.   

I think the main issue is that there was nothing that actually checked the user state before a request was issued.  When I tried to check current_user.current_session_id, sometime it was not set when I thought it should be and vice versa, so I switched to use activate/deactivate()

I'm happy to try to use current_session_id again if you want, but this code review is for "the solution that works" at the moment.  So any mods mean it might go back to not working.  It took a while to get it to work!

Reply:
Okay, gotcha!  Yes, let's focus on fixing the issue and making it work first; we can clean up and improve after where we need to.